src/lib.rs

```diff
@@ -95,6 +95,21 @@ pub trait Integer: Sized + Num + PartialOrd + Ord + Eq {
     /// ~~~
     fn lcm(&self, other: &Self) -> Self;
 
+    /// Greatest Common Divisor (GCD) and
+    /// Lowest Common Multiple (LCM) simultaneously.
+    ///
+    /// More efficient than calling `gcd` and `lcm`
+    /// individually for identical inputs.
+    /// 
+    /// # Examples
+    ///
+    /// ~~~
+    /// # use num_integer::Integer;
+    /// assert_eq!(10.gcd_lcm(&4), (2, 20));
+    /// assert_eq!(8.gcd_lcm(&9), (1, 72));
+    /// ~~~
+    fn gcd_lcm(&self, other: &Self) -> (Self, Self);
```

Comment:
Note that it's a breaking change to add a trait method without a default implementation.  For instance, even `num-bigint` would fail to build with this until it adds the new method.

I think it should be straightforward to implement this generically, although it probably needs `where Self: Clone` to do the math by-value.  When we do implement this for `num-bigint`, we can do so without those external clones, so overall this should be fine.